rs/nervous_system/candid_utils/src/validation.rs

```diff
@@ -70,33 +70,54 @@ impl CandidServiceArgValidationError {
     }
 }
 
+impl std::fmt::Display for CandidServiceArgValidationError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let (kind, message) = self.deconstruct();
+        write!(f, "CandidServiceArgValidationError::{kind}: {message}")
+    }
+}
+
 /// Checks whether `upgrade_args` is a valid argument sequence for `candid_service`.
 ///
-/// Returns the byte encoding of `upgrade_args` in the successful case.
+/// If `upgrade_args` is None, checks that `candid_service` does not require any arguments.
+///
+/// Example `upgrade_args` with two arguments: "((42 : nat32), opt record { foo = opt bar; })".
+///
+/// Returns the byte encoding of `upgrade_args` (if any; otherwise None) in the successful case.
 pub fn validate_upgrade_args(
```

Comment:
Not a new issue in this PR, but...

Taking inspiration from your other function, this should probably also be named `encode_*`. Yes, this performs validation, but the result is not just a yes or no, but rather, a Candid-encoded blob. "validate" under-sells what this does.